main.go

```diff
@@ -134,27 +149,34 @@ func restore_to_dqlite(ep string, dir string) error {
 		LeaderElect: false,
 	}
 	c, err := client.New(etcdcfg)
-	check(err)
+	if err != nil {
+		return err
+	}
 	defer c.Close()
 
 	var i = 0
-	for true {
-		var keyfile = db + "/" + strconv.Itoa(i) + ".key"
-		var datafile = db + "/" + strconv.Itoa(i) + ".data"
+	for {
+		keyfile, datafile := get_kd_files(i)
 		_, err := os.Stat(keyfile)
 		if os.IsNotExist(err) {
 			fmt.Printf("Processed %d entries", i)
 			break
 		}
 
 		keybytes, err := ioutil.ReadFile(keyfile)
-		check(err)
+		if err != nil {
+			return err
+		}
 		key := string(keybytes)
 		databytes, err := ioutil.ReadFile(datafile)
-		check(err)
+		if err != nil {
+			return err
+		}
 
 		logrus.Debugf("%d) %s", i, key)
-		put_key(c, ctx, key, databytes)
+		if err := put_key(c, ctx, key, databytes); err != nil {
+			return err
+		}
```

Comment:
Previously, this would not stop in case of error, but rather proceed. I think it's probably best to avoid any breaking changes, so let's instead do:

```suggestion
		if err := put_key(c, ctx, key, databytes); err != nil {
	             logrus.Errorf("Failed to put key %s: %q", key, err)
		}
```